frontend/src/hooks/useAuth.ts

```diff
@@ -33,12 +33,11 @@ export const useAuth = () => {
 
 export const useLogoutRedirection = () => {
   const router = useRouter();
-  const pathname = usePathname();
   const hasPublicPath = PUBLIC_PATHS.includes(router.pathname);
   const logoutRedirection = async (fullReload: boolean = false) => {
     if (!hasPublicPath) {
       const redirectUrl = `/${RouterType.LOGIN}?redirect=${encodeURIComponent(
-        pathname,
+        router.pathname,
```

Comment:
Hi @abdou6666, i think that the use of next/router will cause an issue especially for **Dynamic Routes**.
For example to to reproduce the bug
- Open Visual editor page 
- Create a new Flow and navigate to this flow
- Click logout button (notice that the ID is missed in the URL)
- Connect again 
- Notice that the redirection is not to the previous visual editor because the dynamic ID is missed